build: automate deploy

samsiegart opened the pull request:
fixes https://github.com/Agoric/wallet-app/issues/40

Looking at the whole tree is probably clearer than the diff: https://github.com/Agoric/wallet-app/tree/automate-deploy

See cloudflare deployment working below

samsiegart commented on package.json:
```diff
@@ -1,199 +1,90 @@
+  ],
+  "scripts": {
+    "build": "rm -rf build && yarn build:wallet && rollup -c && cp _redirects build",
+    "build:wallet": "rm -rf public/wallet && mkdir public/wallet && yarn workspaces run build && cp -r wallet/build/* public/wallet",
```
This is the crux of the change

turadg commented on .github/workflows/test.yml:
```diff
@@ -15,9 +15,8 @@ jobs:
       - uses: actions/setup-node@v3
         with:
-          node-version: '14'
+          node-version: '18.x'
```
consider `latest`. the build tools will always work on LTS releases.
https://github.com/actions/setup-node#supported-version-syntax

turadg commented on .gitignore:
```diff
@@ -1,24 +1,61 @@
-yarn-debug.log*
-yarn-error.log*
+/packages.png
+packages/ui-components/compiled
+packages/ui-components/dist
```
any reason not to ignore all `complied` and `dist`?

turadg commented on README.md:
```diff
@@ -1,45 +1,15 @@
-The UI deps and scripts were bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
+## Production
+### Manual testing
+`yarn && yarn build && yarn preview`
```
consider `yarn install` for clarity

turadg commented on components/install-ses-lockdown.js:
```diff
@@ -0,0 +1,19 @@
+import '@agoric/eventual-send/shim';
+
+lockdown({
+  errorTaming: 'unsafe',
```
worth a comment on why this is okay here

turadg commented on package.json:
```diff
@@ -1,199 +1,90 @@
-  "author": "Agoric",
-  "license": "Apache-2.0",
-  "homepage": "/wallet",
+  "name": "@agoric/wallet",
```
more specifically,
```suggestion
  "name": "@agoric/wallet-site",
```

turadg commented on package.json:
```diff
@@ -1,199 +1,90 @@
     "access": "public"
+  },
+  "engines": {
+    "node": ">=11.0"
```
this is a private package so let's make maintenance easier on ourselves and use the active LTS.
```suggestion
    "node": ">=18.0"
```

turadg commented on wallet/.gitignore:
```diff
@@ -0,0 +1,24 @@
+
+# production
+/build
+/public/lockdown.umd.js
```
this one got in. please do remove it and massage gitignore so it doesn't get in again

turadg commented on wallet/package.json:
```diff
@@ -0,0 +1,200 @@
+{
+  "name": "@agoric/wallet-ui",
```
We've also been calling this the "wallet app". wallet.agoric.app is where you get it.

"wallet-ui" or "wallet-app" both work for me but whichever you pick please ensure the workspace (parent path) matches the name here.

turadg commented on wallet/src/components/SmartWalletConnection.tsx:
```diff
@@ -40,19 +40,21 @@ export const useProvisionPoolMetrics = (unserializer, leader) => {
     let cancelled = false;
     const fetchData = async () => {
-      const follower = await makeFollower(
+      const follower = await makeFollower<{ value: ProvisionPoolMetrics }>(
```
nice

turadg commented on wallet/src/util/keyManagement.ts:
```diff
@@ -250,6 +250,7 @@ export const makeBackgroundSigner = async ({
     granter: string,
     rpcClient: import('@cosmjs/tendermint-rpc').Tendermint34Client,
   ): Promise<GenericAuthorization[]> => {
+    // @ts-expect-error
```
curious what this is. consider some prose explaining the need to suppress.

turadg commented on wallet/src/util/querySwingsetParams.ts:
```diff
@@ -18,6 +18,16 @@ export const querySwingsetParams = async (
   const http = new HttpClient(endpoint);
   const trpc = await Tendermint34Client.create(http);
+  /**
+   * XXX
```
oh is this what the other ones are too? consider filing a ticket in this repo to fix that and linking to it at the sites. second thought, probably not worth the overhead.

turadg reviewed (approved):
Glorious!

Please look over the suggestions and requests before merging. Approving now because I'm so excited to see 58 closed  😸

samsiegart commented:
I tried addressing some feedback but something broke the cloudflare build and I don't have access to the logs. I'll just merge this so we can tweak from a clean slate.
